### PR TITLE
Rerun attempt step benchmark (Exp 1)

### PR DESCRIPTION
**Pull request by pnkraemer**
This PR reruns the attempt step benchmark up to d=16mio. 

It also refactors the plotting code for figure 1.

### CODE REVIEW

File: experiments/1_sterilised_problem.py [hunk 1]
```diff
@@ -184,26 +184,31 @@ def _aslist(arg):
...
+    8192*2,
     8192,
     2048,
     512,
     128,
     32,
     8,
 )
-JIT = (True,)
+JIT = (False,)
```
Comment (pnkraemer) [inline experiments/1_sterilised_problem.py:213]: stuff is jitted already, no point in doing it again (which takes ages anyway)

File: src/hose/plotting.py [hunk 2]
```diff
@@ -321,79 +345,88 @@ def add_legend(ax, fig):
...
+
     fig, axes = plt.subplots(
-        2, len(idxs), figsize=figure_size, sharex="all", sharey="all"
+        figsize=figure_size,
+        nrows=2,
+        ncols=4,
+        dpi=300,
+        sharex=True,
+        sharey=True,
+        constrained_layout=True,
```
Comment (pnkraemer) [inline src/hose/plotting.py:376]: constrained layout is stricter than tight layout

File: src/hose/plotting.py [hunk 3]
```diff
@@ -321,79 +345,88 @@ def add_legend(ax, fig):
...
     )
-    for ax, i, t in zip(axes[0], idxs, _ts):
-        cm1 = ax.imshow(
-            y_means[i][:d].reshape(_d, _d),
+
+    # First row: plot means
+    for (t, y, ax) in zip(ts, y_means, axes[0]):
+        mean_map = ax.imshow(
+            jnp.flip(y[:num_pts].reshape(num_x_points, num_x_points), axis=0),
```
Comment (pnkraemer) [inline src/hose/plotting.py:382]: flip bc for some reason, the y-axis went from 1 to 0. Therefore I flipped the entries and manually changes the y labels (the stuff is symmetric, so the plot looks the same, but flipping felt right)

File: experiments/1_sterilised_problem.py [hunk 4]
```diff
@@ -184,26 +184,33 @@ def _aslist(arg):
...
+    8388608,
+    4194304,
+    2097152,
+    1048576,
+    524288,
+    262144,
+    131072,
+    65536,
+    32768,
+    16384,
```
Comment (nathanaelbosch) [inline experiments/1_sterilised_problem.py:205]: Is there a logic to these numbers? If yes, maybe a formula would be nicer^^ But this can also be done when we re-run things to adjust the plots or something.

File: experiments/1_sterilised_problem.py (hunk 1 above)
Comment (nathanaelbosch) [reply experiments/1_sterilised_problem.py:213]: Should we instead rather remove `JIT` and the code that interacts with it?

**Review by nathanaelbosch — APPROVED** [review]

File: experiments/1_sterilised_problem.py (hunk 4 above)
Comment (pnkraemer) [reply experiments/1_sterilised_problem.py:205]: it is powers of 2... I can rewrite the code later today if you like. For now, though, my brain is somewhere else... :)

File: experiments/1_sterilised_problem.py (hunk 1 above)
Comment (pnkraemer) [reply experiments/1_sterilised_problem.py:213]: this might be a good idea
